### ISSUE TITLE
Better changelog / release notes support

### ISSUE THREAD

**Issue by jpnelson** [atlassian/lerna-semantic-release#52]
Currently the changelog is generated as a file in the repository, which complicates the release process – to get the changelog file published to NPM correctly, we need to do the `post` step *before* the `perform` step, which is not how it should be.

Ideally instead, we would be able to publish release notes summaries instead. Github has a releases API which we could support, and to enable bitbucket support, we can make it agnostic to where it posts these changelogs to – it could be bitbucket snippets, or a third party changelog hosting service.

**Comment by vovacodes** [comment]
I don't think the current approach is bad, the naming is simply not appropriate (`post`). I would call it `generate-changelog`. All this pre-perform-post are too opinionated IMO. Adding Releases API support is a good thing though, but as an extra, not instead.

**Comment by jpnelson** [comment]
I agree, particularly since the current recommended order is `pre` / `post` / `perform` – I'd like to stick closely to how semantic-release does the naming if that is possible somehow

**Comment by vovacodes** [comment]
It is nice to have some predefined set of actions like `pre` / `post` / `perform`, but the reality is that the environments people work in are different so release process might divert a lot from team to team. The approach `semantic-release` takes seems to be too opinionated to me. That is the reason why they still cannot work with GitLab or Bitbucket. I would rather have a set of "semantic" tools for `lerna` that I could use to assemble my release process from. I like the way [conventional-changelog](https://github.com/conventional-changelog) is organized. They have a bunch of tools (changelog generation, github releases, version bump, etc.) each of which has a narrow scope of responsibility that allows them to be easily composed together for a precisely tailored release process. @jpnelson what is your opinion on this?
